test/with_dummyserver/test_socketlevel.py

```diff
@@ -1491,6 +1492,63 @@ def socket_handler(listener: socket.socket) -> None:
                 pool.request("GET", "/", retries=False, timeout=LONG_TIMEOUT)
         assert server_closed.wait(LONG_TIMEOUT), "The socket was not terminated"
 
+    # SecureTransport can read only small pieces of data at the moment.
+    # https://github.com/urllib3/urllib3/pull/2674
+    @notSecureTransport()
+    @pytest.mark.skipif(
+        os.environ.get("CI") == "true" and sys.implementation.name == "pypy",
+        reason="too slow to run in CI",
+    )
+    @pytest.mark.parametrize(
+        "preload_content,read_amt", [(True, None), (False, None), (False, 2**31)]
+    )
+    def test_requesting_large_resources_via_ssl(
```

Comment:
Can we try to manually run PyPy on a `sys.version_info` version less than 3.10 to see that our implementation isn't super slow like it is in the test suite? Would be good to have a comparison between using urllib3+PyPy on a response that's size 2**31-1 and 2**31

Reply:
It took 5 minutes 27 seconds for PyPy 3.9 to run the three tests. When I changed `2**31` to `2**31 - 100` (this plus the headers length end ups being near the limit) in the test, it took 5 minutes and 16 seconds.

Reply:
Great, thank you!